bin/opm

```diff
@@ -2195,6 +2198,57 @@ sub rebase_path ($$$) {
     return undef;
 }
 
+sub do_clean {
+    my @valid_options = ('dist');
+    
+    if (@_ == 0) {
+        err "no clean option specified.\n";
+    }
+    
+    if ($_[0] eq 'dist') {
+        my $dist_file = "dist.ini";
+        my $data = read_ini($dist_file);
+        my $default_sec = $data->{default};
+        my $dist_name = $default_sec->{name};
+        if (!$dist_name) {
+            err "$dist_file: key \"name\" not found in the default section.\n";
+        }
+
+        if (length $dist_name < 3
+            || $dist_name =~ /[^-\w]|^(?:nginx|luajit|resty|openresty|opm|restydoc.*|ngx_.*|.*-nginx-module)$/i)
+        {
+            err "$dist_file: bad dist name: $dist_name\n";
+        }
+        opendir(my $dh, './') 
+            or err "failed to open directory './'!\n";
```

Comment:
We need to use `: $!` instead of `!` here. See other similar places. Be careful ;)